nnpdf_data/nnpdf_data/commondata/ATLAS_Z0J_8TEV/filter_utils.py

```diff
@@ -0,0 +1,284 @@
+import numpy as np
+import yaml
+import pandas as pd
+import logging
+import os
+
+RAP_KINEMATICS_BINS = {
+  "Table29" : [0.0, 0.4],
+  "Table30" : [0.4, 0.8],
+  "Table31" : [0.8, 1.2],
+  "Table32" : [1.2, 1.6],
+  "Table33" : [1.6, 2.0],
+  "Table34" : [2.0, 2.4],
+}
+
+M_KINEMTAICS_BINS = {
+  "Table35" : [12.0, 20.0],
+  "Table36" : [20.0, 30.0],
+  "Table37" : [30.0, 46.0],
+  "Table38" : [46.0, 66.0],
+  "Table39" : [66.0, 116.0],
+  "Table40" : [116.0, 150.0]
+}
+
+ATLASLUMI12 = 2.8
+
+SQRTS = 8000
+
+class Extractor:
+  def __init__(self, metadata_file, observable, mult_factor):
+
+    # Open metadata and select process
+    with open(metadata_file, 'r') as file:
+      metadata = yaml.safe_load(file)
+      self.metadata = next(
+        (md for md in metadata["implemented_observables"] if md['observable_name'] == observable),
+        None
+        )
+      
+    # Initialise dcit of tables
+    self.tables = {}
+    self.observable = observable
+    self.mult_factor = mult_factor
+      
+    # Select the bins for the second kinematic variable
+    if observable == 'PT-Y':
+      self.kin2_dict = RAP_KINEMATICS_BINS
+      self.kin_labels =['y', 'pT', 'sqrts']
+    elif observable == 'PT-M':
+      self.kin2_dict = M_KINEMTAICS_BINS
+      self.kin_labels =['m', 'pT', 'sqrts']
+    else:
+      raise Exception(f"Observable {observable} not listed in the metadata file.")
+    
+
+  def __extract_kinematics(self, table: dict,
+                         tab_number: int):
+    data = table['independent_variables'][0]
+    label = self.kin_labels
+    kinematics = []
+    for bin in data['values']:
+      kin_min = bin['low']
+      kin_max = bin['high']
+      kin2_bin = self.kin2_dict[f'Table{tab_number}']
+      kin_bin = {
+        label[0]: {'min': kin2_bin[0],
+                  'mid': None,
+                  'max': kin2_bin[1]},
+        label[1]: {'min': kin_min, 
+                  'mid': None, 
+                  'max': kin_max},
```

Comment:
You should also fill the mid-values. Actually, `mid` is the field that is used everywhere in the codes.

Reply:
I think (if not, it would be a nice feature to have) that I added in the parser that if the mid is None it gets automatically filled _by the parser_

(of course, given that we technically know it at this point we might also fill it here)